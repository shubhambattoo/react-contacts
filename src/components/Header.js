import { useState } from 'react';
import { connect } from 'react-redux';
import { setFilter } from '../redux/actions/searchFilter';

export const Header = ({ setFilter }) => {
  const [search, setSearch] = useState('');

  function handleSerach(e) {
    e.preventDefault();
    setFilter(search);
    setSearch('');
  }

  return (
    <header>
      <nav className="navbar navbar-expand-lg navbar-light bg-light">
        <a className="navbar-brand" href="/">
          Contacts App
        </a>

        <form
          className="form-inline ml-auto my-2 my-lg-0"
          onSubmit={handleSerach}
        >
          <input
            className="form-control mr-sm-2"
            type="search"
            placeholder="Search"
            aria-label="Search"
            value={search}
            onChange={(e) => setSearch(e.target.value)}
          />
          <button
            className="btn btn-outline-success my-2 my-sm-0"
            type="submit"
          >
            Search
          </button>
        </form>
      </nav>
    </header>
  );
};

export default connect(null, { setFilter })(Header);
