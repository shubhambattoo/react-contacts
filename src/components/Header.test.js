import React from 'react';
import { shallow, mount } from 'enzyme';
import { Header } from './Header';
import renderer from 'react-test-renderer';

describe('HeaderComponent', () => {
  const wrapper = shallow(<Header />);

  it('should have brand title', () => {
    const tag = wrapper.find('nav a');
    const title = 'Contacts App';
    expect(tag.text()).toEqual(title);
  });

  it('should update the text on input', () => {
    const wrapper = mount(<Header />);
    const input = wrapper.find('form input');
    input.simulate('change', { target: { value: 'shubham' } });
    expect(input.instance().value).toBe('shubham');
  });

  test('snapshot renders', () => {
    const component = renderer.create(<Header />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
