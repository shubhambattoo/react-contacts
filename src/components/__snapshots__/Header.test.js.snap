// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent snapshot renders 1`] = `
<header>
  <nav
    className="navbar navbar-expand-lg navbar-light bg-light"
  >
    <a
      className="navbar-brand"
      href="/"
    >
      Contacts App
    </a>
    <form
      className="form-inline ml-auto my-2 my-lg-0"
      onSubmit={[Function]}
    >
      <input
        aria-label="Search"
        className="form-control mr-sm-2"
        onChange={[Function]}
        placeholder="Search"
        type="search"
        value=""
      />
      <button
        className="btn btn-outline-success my-2 my-sm-0"
        type="submit"
      >
        Search
      </button>
    </form>
  </nav>
</header>
`;
